### PR TITLE
feat: commitlint setup [#18]

### PR DESCRIPTION
**Pull request by anjaliahlawat**

### CODE REVIEW

**Review by h2grover — CHANGES_REQUESTED** [review]
@anjaliahlawat I think we are not on the same page, can you please let me know why I asked you to install husky and commitLint ? And is the usage of these packages?

**Comment by anjaliahlawat** [comment]
@h2grover I think they help in improving and linting the commit messages.

**Comment by anjaliahlawat** [comment]
@h2grover https://commitlint.js.org/#/guides-ci-setup. This link is for linting CI servers. Shall I do this setup as well?

**Comment by anjaliahlawat** [comment]
@h2grover  I guess we are trying to do something like this - https://www.freecodecamp.org/news/how-to-add-commit-hooks-to-git-with-husky-to-automate-code-tasks/

**Comment by h2grover** [comment]
@anjaliahlawat 

> @h2grover https://commitlint.js.org/#/guides-ci-setup. This link is for linting CI servers. Shall I do this setup as well?

No

**Comment by h2grover** [comment]
> @h2grover I guess we are trying to do something like this - https://www.freecodecamp.org/news/how-to-add-commit-hooks-to-git-with-husky-to-automate-code-tasks/

yes, so how are you achieving this with your code base?


Please run commitLint in pre-commit husky hook

**Comment by anjaliahlawat** [comment]
I observed that in pre-commit hooks, last commit message was linted but when I used commit-msg hooks, with the commitlint command it was linting current message. Is this behaviour alright?

File: package.json [hunk 1]
```diff
@@ -2,12 +2,34 @@
...
+    "postinstall": "husky install"
   },
-  "dependencies": {
+  "husky": {
+    "hooks": {
+      "pre-commit": "echo checking commit",
+      "commit-msg": "commitlint --edit $HUSKY_GIT_PARAMS"
+    }
+  },
+  "commitlint": {
```
Comment (h2grover) [inline package.json]: Let's add these rules in the `commitlint.config.js` file

File: package.json [hunk 2]
```diff
@@ -2,12 +2,34 @@
...
+    "rules": {
+      "references-empty": [2, "never"]
+    },
+    "parserPreset": {
+      "parserOpts": {
+        "issuePrefixes": ["[#]"]
```
Comment (h2grover) [inline package.json]: We should not enforce this, `"references-empty": [2, "never"]` is sufficient.

File: package.json [hunk 3]
```diff
@@ -2,12 +2,34 @@
...
+      }
+    }
+  },
+  "dependencies": {   
     "@getgauge/cli": "^1.1.7",
-    "dotenv": "^8.2.0",
+    "dotenv": "^8.2.0",
```
Comment (h2grover) [inline package.json]: Remove whitespace.

File: package.json [hunk 4]
```diff
@@ -2,12 +2,34 @@
   "name": "gauge-js-template",
   "description": "Starter template for writing JavaScript tests for Gauge",
   "scripts": {
-    "test": "./node_modules/.bin/gauge run -p specs/"
+    "test": "./node_modules/.bin/gauge run -p specs/",
+    "postinstall": "husky install"
```
Comment (h2grover) [inline package.json]: what is this for? I guess it is not needed plz remove it.

File: package.json [hunk 5]
```diff
@@ -2,12 +2,34 @@
...
   },
-  "dependencies": {
+  "husky": {
+    "hooks": {
+      "pre-commit": "echo checking commit",
```
Comment (h2grover) [inline package.json]: Why outputting on console? Please remove.

File: package.json [hunk 6]
```diff
@@ -2,12 +2,34 @@
...
+      "commit-msg": "commitlint --edit $HUSKY_GIT_PARAMS"
+    }
+  },
+  "commitlint": {
+    "rules": {
+      "references-empty": [2, "never"]
```
Comment (h2grover) [inline package.json]: lets add https://commitlint.js.org/#/reference-rules?id=type-enum also

**Review by h2grover — CHANGES_REQUESTED** [review]

**Review by h2grover — COMMENTED** [review]
LGTM 👍  but can you please update the PR title manually, it should have same format as commit.

**Comment by anjaliahlawat** [comment]
@h2grover Please check the pr title. If the format's fine, I will merge this pull request.

**Comment by h2grover** [comment]
@anjaliahlawat I have updated the title, the first letter should not be capatial. 

> @h2grover Please check the pr title. If the format's fine, I will merge this pull request.

This is your repo so you have merging rights but please do not merge the PR, that is my job to do.

**Review by h2grover — APPROVED** [review]

**Pull request merged**